tcms/testplans/views.py

```diff
@@ -253,28 +254,22 @@ def calculate_stats_for_testplans(plans):
     return plans
 
 
-def get(request,  # pylint: disable=missing-permission-required
-        plan_id, slug=None, template_name='plan/get.html'):
+class TestPlanGetView(DetailView):  # pylint: disable=missing-permission-required
     """Display the plan details."""
 
-    try:
-        test_plan = TestPlan.objects.select_related().get(plan_id=plan_id)
-    except ObjectDoesNotExist:
-        raise Http404
-
-    if slug is None:
-        return HttpResponsePermanentRedirect(reverse('test_plan_url',
-                                                     args=[plan_id, slugify(test_plan.name)]))
+    template_name = 'plan/get.html'
+    http_method_names = ['get']
+    model = TestPlan
+    pk_url_kwarg = 'plan_id'
 
-    # Initial the case counter
-    confirm_status_name = 'CONFIRMED'
-    test_plan.run_case = test_plan.case.filter(case_status__name=confirm_status_name)
-    test_plan.review_case = test_plan.case.exclude(case_status__name=confirm_status_name)
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        confirm_status_name = 'CONFIRMED'
+        self.object.run_case = self.object.case.filter(case_status__name=confirm_status_name)
+        self.object.review_case = self.object.case.exclude(case_status__name=confirm_status_name)
```

Comment:
`review_case.count` is the only thing used in template. Convert to context variable.

Reply:
still not converted